applications/luci-app-shadowsocks-libev/htdocs/luci-static/resources/view/shadowsocks-libev/servers.js

```diff
@@ -16,6 +17,42 @@ return L.view.extend({
 
 		s = m.section(form.GridSection, 'server');
 		s.addremove = true;
+		s.renderSectionAdd = function(extra_class) {
+			var el = form.GridSection.prototype.renderSectionAdd.apply(this, arguments);
+			el.appendChild(E('button', {
+				'class': 'cbi-button cbi-button-add',
+				'title': _('Import Links'),
+				'click': L.ui.createHandlerFn(this, function() {
+					return new Promise(function(resolveFn, rejectFn) {
+						var textarea = new L.ui.Textarea();
+						L.ui.showModal(_('Import Links'), [
+							textarea.render(),
+							E('div', { class: 'right' }, [
+								E('button', {
+									class: 'btn',
+									click: function() {
+										L.ui.hideModal();
+										rejectFn();
+									}
+								}, [ _('Cancel') ]),
+								' ',
+								E('button', {
+									class: 'btn cbi-button-action',
+									click: function() {
+										textarea.getValue().split('\n').filter(function(s) {
+											return !!s;
+										}).forEach(function(s) {
+											debugger;
+										});
+									}
+								}, [ _('Import') ])
+							])
+						]);
+					});
+				})
+			}, [ _('Import Links') ]));
+			return el;
```

Comment:
You don't call `resolveFn` anywhere here, that is probably not intended.

Reply:
It's working in progress.

Reply:
Ok :)